mobify/platform-scaffold
# preview build saves files

By default, the `webpack-dev-server` serves files from memory, so they're not visible in the file system. This PR adds the `write-file-webpack-plugin` to save the packed files in the `build` directory. It also adds `Cache-Control: no-cache` to all served files to avoid source maps being cached in Chrome.

 **JIRA**: N/A

## Changes
- Add `write-file-webpack-plugin` and configure it in `web/dev-server/index.js`
- Add cache control in `web/dev-server/index.js

## Todos

## How to test-drive this PR
- Run `npm start`
- When the bundle is valid, check the `build` directory - the packed files, plus gzipped versions, plus source maps, should all be present
- Use `wget -O /dev/null --server-response --no-check-certificate --header 'accept-encoding: gzip' https://localhost:8443/loader.js.map` to verify that the response contains `Cache-Control: no-cache`
- Use `npm run test:pwa-preview` to verify that the lighthouse score is unaffected
- Use preview to verify that it works
